ydb/library/yql/yt/native/plugin.cpp

```diff
@@ -554,10 +622,18 @@ class TYqlPlugin
     std::optional<TString> DefaultCluster_;
     THashMap<TString, TString> Modules_;
     TYsonString OperationAttributes_;
+    TString YqlAgentToken_;
+
     YT_DECLARE_SPIN_LOCK(NThreading::TReaderWriterSpinLock, ProgressSpinLock);
     THashMap<TQueryId, TActiveQuery> ActiveQueriesProgress_;
     TVector<NYql::TDataProviderInitializer> DataProvidersInit_;
 
+    void RemoveQuery(TQueryId queryId)
+    {
+        auto guard = WriterGuard(ProgressSpinLock);
+        ActiveQueriesProgress_.erase(queryId);
```

Comment:
We didn't have lock between checking `contains` and this code. Someone else may have acquired the lock and erased the query.
I am not sure how erase works if there is no such element. I think it throws some exception
```suggestion
        if (ActiveQueriesProgress_.contains(queryId)) {
            ActiveQueriesProgress_.erase(queryId);
        }
```

Reply:
Erase does not fail if there is no query. Therefore, I thought that this was not necessary.

But for clarity of the code, put it here